### ISSUE TITLE
Validate device addresses better

### ISSUE THREAD

**Issue by Catfriend1** [Catfriend1/syncthing-android#264]
Quote from @steventebrinke (https:// github.com/syncthing/syncthing-android/issues/1268):

> When I started using syncthing (v0.14.51), I couldn't get the sync working, the UI just said that the server was offline. It turned out that I'd incorrectly entered myserver.example.org instead of tcp://myserver.example.org. The UI shows no information about this (only the debug log was helpful). It would be nice if the UI doesn't allow saving invalid addresses. IMO any address that gives a parsing error should result in reporting an error to the user.

**Comment by Catfriend1** [comment]
From the docs at https://docs.syncthing.net/users/config.html#device-element :

```
address
Contains an address or host name to use when attempting to connect to this device. Entries other than dynamic must be prefixed with tcp:// (dual-stack), tcp4:// (IPv4 only) or tcp6:// (IPv6 only). Note that IP addresses need not use tcp4/tcp6; these are optional. Accepted formats are:

IPv4 address (tcp://192.0.2.42)
The default port (22000) is used.
IPv4 address and port (tcp://192.0.2.42:12345)
The address and port is used as given.
IPv6 address (tcp://[2001:db8::23:42])
The default port (22000) is used. The address must be enclosed in square brackets.
IPv6 address and port (tcp://[2001:db8::23:42]:12345)
The address and port is used as given. The address must be enclosed in square brackets.
Host name (tcp6://fileserver)
The host name will be used on the default port (22000) and connections will be attempted only via IPv6.
Host name and port (tcp://fileserver:12345)
The host name will be used on the given port and connections will be attempted via both IPv4 and IPv6, depending on name resolution.
dynamic
The word dynamic (without tcp:// prefix) means to use local and global discovery to find the device.
You can set multiple addresses and combine it with the dynamic keyword for example:

<device id="...">
    <address>tcp://192.0.2.1:22001</address>
    <address>tcp://192.0.1.254:22000</address>
    <address>dynamic</address>
</device>
```

**Comment by Catfriend1** [comment]
The fix will be part of an upcoming release.

**Comment by Catfriend1** [comment]
@steventebrinke Can you confirm this is fixed in the most recent release 1.0.0.19?
